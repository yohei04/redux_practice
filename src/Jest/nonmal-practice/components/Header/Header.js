import React from 'react';

const Header = ({ text }) => {
  return (
    <div>
      <h1 data-testid="h1tag" className="fancy-h1">
        {text}
      </h1>
    </div>
  );
};

export default Header;
