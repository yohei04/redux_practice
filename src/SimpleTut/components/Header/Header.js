import React from 'react';
import './styles.scss';
import Logo from '../../assets/graphics/logo.png';

const Header = (props) => {
  return (
    <header data-testid='headerComponent'>
      <div className="wrap">
        <div className="logo">
          <img src={Logo} alt="Logo" data-testid='logoIMG'/>
        </div>
      </div>
    </header>
  );
};

export default Header;
