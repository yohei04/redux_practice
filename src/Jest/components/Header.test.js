import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from './Header';

it('renders', () => {
  const { asFragment } = render(<Header text="Hello!" />);
  expect(asFragment()).toMatchSnapshot();
});
