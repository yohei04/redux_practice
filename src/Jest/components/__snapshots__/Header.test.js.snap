// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Hello!
    </h1>
  </div>
</DocumentFragment>
`;
